umm-spec-lib/src/cmr/umm_spec/field_update.clj

```diff
@@ -34,9 +34,12 @@
 (defmethod apply-umm-list-update :find-and-remove
   [update-type umm update-field update-value find-value]
   (if (seq (get-in umm update-field))
-    (update-in umm update-field #(remove (fn [x]
-                                           (value-matches? find-value x))
-                                         %))
+    (let [umm-updated (update-in umm update-field #(remove (fn [x]
+                                                             (value-matches? find-value x))
+                                                     %))]
+      (if (seq (get-in umm-updated update-field))
+        umm-updated
+        (update-in umm-updated update-field seq)))
```

Comment:
I thought I could just do (remove-map-keys empty? umm-updated) to get rid of the empty platforms/instruments and so on. But when I did that, bulk update status returns things like "No matching clause" which I still don't know what exactly it means. The strange part is, even when I only remove partial of the platforms, it still gives me that status. I don't know if it's because I removed some other nil fields that somehow matter to this. Looks like it needs all the required elements there, so I just made them nil.